opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -1543,6 +1543,32 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
     static bool isInitialized_;
 };
 
+#define BOFS(T) BlackOilFluidSystem<T, BlackOilDefaultIndexTraits>
+
+#define DECLARE_INSTANCE(T) \
+template<> unsigned char BOFS(T)::numActivePhases_; \
+template<> std::array<bool, BOFS(T)::numPhases> BOFS(T)::phaseIsActive_; \
+template<> std::array<short, BOFS(T)::numPhases> BOFS(T)::activeToCanonicalPhaseIdx_; \
+template<> std::array<short, BOFS(T)::numPhases> BOFS(T)::canonicalToActivePhaseIdx_; \
+template<> double BOFS(T)::surfaceTemperature; \
+template<> double BOFS(T)::surfacePressure; \
+template<> double BOFS(T)::reservoirTemperature_; \
+template<> bool BOFS(T)::enableDissolvedGas_; \
+template<> bool BOFS(T)::enableDissolvedGasInWater_; \
+template<> bool BOFS(T)::enableVaporizedOil_; \
+template<> bool BOFS(T)::enableVaporizedWater_; \
+template<> bool BOFS(T)::enableDiffusion_; \
+template<> std::shared_ptr<OilPvtMultiplexer<T>> BOFS(T)::oilPvt_; \
+template<> std::shared_ptr<GasPvtMultiplexer<T>> BOFS(T)::gasPvt_; \
+template<> std::shared_ptr<WaterPvtMultiplexer<T>> BOFS(T)::waterPvt_; \
+template<> std::vector<std::array<T, 3>> BOFS(T)::referenceDensity_; \
+template<> std::vector<std::array<T, 3>> BOFS(T)::molarMass_; \
+template<> std::vector<std::array<T, 9>> BOFS(T)::diffusionCoefficients_; \
+template<> bool BOFS(T)::isInitialized_;
+
+//DECLARE_INSTANCE(float)
+DECLARE_INSTANCE(double)
```

Comment:
as this is a header pulled in all over the place please undef the macro when done.